proselint/tools.py

```diff
@@ -230,6 +231,48 @@ def line_and_column(text, position):
     return (line_no, position - position_counter)
 
 
+def find_ignored_checks(available_checks, text):
+    ignored_checks = collections.defaultdict(dict)
+    for match in re.finditer(r"^[!-/:-@[-`{-~\s]*proselint: (?P<action>disable|enable)(?P<checks>(?:=(?:[\w\.,]*)))?", text, flags=re.MULTILINE):
```

Comment:
Consider the computational cost of this. RegExes are expensive and moderately slow, so running one on every line just to determine whether or not we can disable some checks feels inefficient.

I am not quite sure what the solution to this would be yet, but in the meantime any ideas are welcome. They may not be quite as simple or elegant, but proselint's performance is lacking as it stands, and implementing this would be a major blow in the wrong direction.